cyclonedx/model/tool.py

```diff
@@ -0,0 +1,357 @@
+    def hashes(self, hashes: Iterable[HashType]) -> None:
+        self._hashes = SortedSet(hashes)
+
+    @property
+    @serializable.view(SchemaVersion1Dot4)
+    @serializable.view(SchemaVersion1Dot5)
+    @serializable.view(SchemaVersion1Dot6)
+    @serializable.xml_array(serializable.XmlArraySerializationType.NESTED, 'reference')
+    @serializable.xml_sequence(5)
+    def external_references(self) -> 'SortedSet[ExternalReference]':
+        """
+        External References provides a way to document systems, sites, and information that may be relevant but which
+        are not included with the BOM.
+
+        Returns:
+            Set of `ExternalReference`
+        """
+        return self._external_references
+
+    @external_references.setter
+    def external_references(self, external_references: Iterable[ExternalReference]) -> None:
+        self._external_references = SortedSet(external_references)
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Tool):
+            return hash(other) == hash(self)
+        return False
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, Tool):
+            return _ComparableTuple((
+                self.vendor, self.name, self.version
+            )) < _ComparableTuple((
+                other.vendor, other.name, other.version
+            ))
+        return NotImplemented
+
+    def __hash__(self) -> int:
+        return hash((self.vendor, self.name, self.version, tuple(self.hashes), tuple(self.external_references)))
+
+    def __repr__(self) -> str:
+        return f'<Tool name={self.name}, version={self.version}, vendor={self.vendor}>'
+
+
+class ToolsRepository:
+    """
+    The repository of tool formats
+
+    This is done so we can maintain backward-compatibility with CycloneDX <= 1.4
+    If using a SortedSet of tools, the object will behave like a sorted set of
+    tools. Otherwise, it will behave like an object with `components`
+    and `services` attributes (which are SortedSets of their respective types).
+    """
+    def __init__(self, *, components: Optional[Iterable[Component]] = None,
+                 services: Optional[Iterable[Service]] = None,
+                 # Deprecated in v1.5
+                 tools: Optional[Iterable[Tool]] = None) -> None:
+
+        if tools and (components or services):
+            # Must use components/services or tools. Cannot use both
+            raise MutuallyExclusivePropertiesException(
+                'Cannot define both old (CycloneDX <= 1.4) and new '
+                '(CycloneDX >= 1.5) format for tools.'
+            )
+
+        if tools:
+            warn('Using Tool is deprecated as of CycloneDX v1.5. Components and Services should be used now. '
+                 'See https://cyclonedx.org/docs/1.5/', DeprecationWarning)
+
+        self._components = SortedSet(components or ())
+        self._services = SortedSet(services or ())
+        self._tools = SortedSet(tools or ())
```

Comment:
I want to use all these methods `SortedSet` has.

As a user: Give me an accessor (getter/setter) to `self._tools` I do not care for all these convenient "backwards-compativle" dunder-methots, they are not complete. 
As a maintainer: i dont want to chase `SortedSet` and re-implement all these methods in the class.
As a user: I want to have well-documented readable methods. I am not oaky with re-routing `__call__` or other dunder-methods from this class to `self_tools` ... my linter hates this!
(reminder: users are software-developers)